extensions/sql-database-projects/src/models/projectEntry.ts

```diff
@@ -44,25 +44,32 @@ export class FileProjectEntry extends ProjectEntry implements IFileProjectEntry
 }
 
 export class DacpacReferenceProjectEntry extends FileProjectEntry implements IDatabaseReferenceProjectEntry {
+	databaseSqlCmdVariableValue?: string;
+	databaseSqlCmdVariableName?: string;
 	databaseVariableLiteralValue?: string;
-	databaseSqlCmdVariable?: string;
-	serverName?: string;
-	serverSqlCmdVariable?: string;
+	serverSqlCmdVariableName?: string;
+	serverSqlCmdVariableValue?: string;
 	suppressMissingDependenciesErrors: boolean;
 
 	constructor(settings: IDacpacReferenceSettings) {
-		super(settings.dacpacFileLocation, '', EntryType.DatabaseReference);
-		this.databaseSqlCmdVariable = settings.databaseVariable;
-		this.databaseVariableLiteralValue = settings.databaseName;
-		this.serverName = settings.serverName;
-		this.serverSqlCmdVariable = settings.serverVariable;
+		super(settings.dacpacFileLocation, /* relativePath doesn't get set for database references */ '', EntryType.DatabaseReference);
 		this.suppressMissingDependenciesErrors = settings.suppressMissingDependenciesErrors;
+
+		if (settings.databaseVariable) { // full SQLCMD variable
+			this.databaseSqlCmdVariableName = settings.databaseName;
+			this.databaseSqlCmdVariableValue = settings.databaseVariable;
+		} else { // just a literal
+			this.databaseVariableLiteralValue = settings.databaseName;
+		}
```

Comment:
This branch shouldn't be necessary any longer, now that `IDacpacReferenceSettings` has them as different properties.

Same with `SqlProjectReferenceEntry` down below.